downscale/estimate_forcing_regression.f90

```diff
@@ -151,19 +178,26 @@ end subroutine heapsort
   end interface
   ! =========== end interfaces, start code =============
 
-  real (dp), intent (in) :: x (:, :), z (:, :)  ! station and grid point description arrays
+  real (dp), intent (inout) :: x (:, :), z (:, :)  ! station and grid point description arrays
   real (dp), intent (in) :: maxdistance         ! max distance for weight function
   integer (i4b), intent (in) :: ngrid           ! number of grid points
   real (dp), intent (in) :: times (:)!time step array
 
   ! AWW added next
   integer (i4b), intent (in) :: st_rec, end_rec
 
-  character (len=100), intent (in) :: stnid (:)!station id array
-  character (len=100), intent (in) :: stnvar !control file variables
-  character (len=500), intent (in) :: directory
-  character (len = 500),intent(in) :: gen_sta_weights     ! flag for generating station weight file
-  character (len = 500),intent(in) :: sta_weight_name     ! station weight file name
+  character (len=100), intent (in)  :: stnid (:)!station id array
+  character (len=100), intent (in)  :: stnvar !control file variables
+  character (len=500), intent (in)  :: directory
+  character (len = 500),intent(in)  :: gen_sta_weights     ! flag for generating station weight file
+  character (len = 500),intent(in)  :: sta_weight_name     ! station weight file name
+  character (len = 2000),intent(in) :: nwp_input_list      !file containing list of NWP predictor input files
```

Comment:
here, too, more descriptive names:  instead of 'nwp_input_list', something like 'dyn_grid_file_list' ... 'input' is too generic, 'nwp' too specific.